app/presenters/results/firm_presenter.rb

```diff
@@ -0,0 +1,84 @@
+module Results
+  class FirmPresenter
+    LESS_THAN_FIFTY_K_ID = 1
+
+    DIRECTLY_MAPPED_FIELDS = %i[
+      id
+      registered_name
+      telephone_number
+      website_address
+      email_address
+      free_initial_meeting
+      minimum_fixed_fee
+      other_advice_methods
+      in_person_advice_methods
+      investment_sizes
+      adviser_accreditation_ids
+      adviser_qualification_ids
+      ethical_investing_flag
+      sharia_investing_flag
+      workplace_financial_advice_flag
+      non_uk_residents_flag
+      languages
+      total_advisers
+      total_offices
+    ].freeze
+
+    TYPES_OF_ADVICE_FIELDS = %i[
+      retirement_income_products_flag
+      pension_transfer_flag
+      long_term_care_flag
+      equity_release_flag
+      inheritance_tax_and_estate_planning_flag
+      wills_and_probate_flag
+    ].freeze
+
+    MAPPED_FIELDS = (DIRECTLY_MAPPED_FIELDS + TYPES_OF_ADVICE_FIELDS).freeze
+
+    OTHER_FIELDS = %i[
+      offices
+      advisers
+      closest_adviser_distance
+    ].freeze
+
+    attr_reader(*MAPPED_FIELDS)
+    attr_accessor(*OTHER_FIELDS)
+
+    def initialize(object)
```

Comment:
With these presenters, is there a more specific name we can use for the input variable? Lacking the domain knowledge to make much of a suggestion but wondering if there's something more descriptive than `object`.

Reply:
Good point. It should represent a "hit" or _record_ from Algolia, so it's a hash containing information about a firm/adviser/office. Something like `firm`/`adviser`/`office` might actually work better than a generic `object` indeed. Open to suggestion 👍